appcues/src/main/java/com/appcues/trait/appcues/TooltipsPaths.kt

```diff
@@ -163,89 +198,166 @@ private fun getTooltipPointerPath(
+        targetValue = when (tooltipSettings.tooltipPointerPosition) {
+            Top, Left -> -tooltipSettings.pointerLengthPx
+            Bottom, Right -> tooltipSettings.pointerLengthPx
+            else -> 0f
+        },
+        animationSpec = animationSpec
     )
 
     // vertical pointer implementation, probably in the future make this a separate method and add
     // another one to generate a horizontal pointer for leading and trailing positions.
     return Path().apply {
 
-        TooltipPointerPath(tooltipSettings, pointerOffsetX.second, animationSpec)
-
-        val pathOffsetX = animateFloatAsState(pointerOffsetX.first, animationSpec)
-        val pathOffsetY = if (tooltipSettings.tooltipPointerPosition is Bottom)
-            containerDimens.heightPx - tooltipSettings.pointerLengthPx else tooltipSettings.pointerLengthPx
+        val animatedPointerOffsetX = animateFloatAsState(pointerOffsetX, animationSpec)
+        val animatedPointerOffsetY = animateFloatAsState(pointerOffsetY, animationSpec)
 
-        // Move path to correct place
-        translate(Offset(pathOffsetX.value, pathOffsetY))
+        if (tooltipSettings.tooltipPointerPosition.isVertical) {
+            TooltipVerticalPointerPath(tipLength.value, tooltipSettings.pointerBasePx, verticalTipOffset, animationSpec)
+            translate(Offset(animatedPointerOffsetX.value, pointerOffsetY))
+        } else {
+            TooltipHorizontalPointerPath(tipLength.value, tooltipSettings.pointerBasePx, horizontalTipOffset, animationSpec)
+            translate(Offset(pointerOffsetX, animatedPointerOffsetY.value))
+        }
     }
 }
 
 private fun calculatePointerXOffset(
     containerDimens: TooltipContainerDimens,
     tooltipSettings: TooltipSettings,
     pointerOffsetX: Float,
-    cornerRadius: Float,
+    cornerRadiusPx: Float,
 ): Pair<Float, Float> {
-    val tempPointerOffset = (containerDimens.widthPx / 2) - tooltipSettings.pointerBaseCenterPx + pointerOffsetX
-
     // boundaries for the tooltip
     val minPointerOffset = 0f
-    val minPointerOffsetCornerRadius = minPointerOffset + cornerRadius
+    val minPointerOffsetCornerRadius = minPointerOffset + cornerRadiusPx
     val maxPointerOffset = max((containerDimens.widthPx - tooltipSettings.pointerBasePx), 0f)
-    val maxPointerOffsetCornerRadius = maxPointerOffset - cornerRadius
+    val maxPointerOffsetCornerRadius = maxPointerOffset - cornerRadiusPx
+
+    // figure out the offset of the pointer and offset for the tooltip pointer when hits the edges
+    var pointerTipOffset = 0f
+    var pointerOffset = (containerDimens.widthPx / 2) - tooltipSettings.pointerBaseCenterPx + pointerOffsetX
+
+    when {
+        tooltipSettings.tooltipPointerPosition is Right -> {
+            pointerOffset = containerDimens.widthPx - tooltipSettings.pointerLengthPx
+        }
+        tooltipSettings.tooltipPointerPosition is Left -> {
+            pointerOffset = tooltipSettings.pointerLengthPx
+        }
```

Comment:
I'm a bit lost on what these are doing to be honest

Reply:
ah ok the edge case for when the pointer target element is on the far edge of the screen - all good, maybe just a comment would help remind what case that is handling.